src/yunohost/app.py

```diff
@@ -718,9 +718,38 @@ def app_install(operation_logger, auth, app, label=None, args=None, no_remove_on
         },
     }
 
-    if app in app_list(raw=True) or ('@' in app) or ('http://' in app) or ('https://' in app):
+    def confirm_install(confirm):
+
+        # Ignore if there's nothing for confirm (good quality app), if --force is used
+        # or if request on the API (confirm already implemented on the API side)
+        if confirm is None or force or msettings.get('interface') == 'api':
+            return
+
+        answer = msignals.prompt(m18n.n('confirm_app_install_'+confirm,
+                                   answers='Y/N'))
+        if answer.upper() != "Y":
+            raise MoulinetteError(errno.EINVAL, m18n.n("aborting"))
+
+
+    raw_app_list = app_list(raw=True)
+    if app in raw_app_list or ('@' in app) or ('http://' in app) or ('https://' in app):
```

Comment:
I'm not sure... do we support "git://" style of uri?

Reply:
Uh I dunno, I guess nope, but what do you mean ?

Reply:
That would mean a user could do `yunohost app install git://someurl.com/path/to/repo` (which used to be supported on github) to install an application other git protocol.

Same for `ssh://` uri style.

Technically it should work if we don't have to strong checks in the code.

Reply:
Uh okay, but that's not really the point of this PR, I just used the same piece of code as before, I just factorized the `app_list(raw=True)` ?

Reply:
Ok, let's ignore that then. It was just to had the check here like `('git://' in app)`